File: writeTable.js
const getJSON = require('get-json')
const fs = require('fs')
const path = require('path')
const chalk = require('chalk')
const figlet = require('figlet')
var AWS = require('aws-sdk')

const outputFile = process.argv.slice(2)[0]
const dataDir = 'data/'

console.log(chalk.green(figlet.textSync('Bitcoin Chart Scraper', {
	font: 'Pepper',
	kerning: 'fitted'
})))
console.log(chalk.dim(`Combining data to=${chalk.yellow(outputFile)} \n`))

const files = []

function compare (a, b) {
	if (a.time < b.time) {
		return -1
	}

	if (a.time > b.time) {
		return 1
	}

	return 0
}

fs.readdir(dataDir, (err, list) => {
	list.forEach(file => {
		let date
		let time

		try {
			date = new Date(file.split('bitstampUSD-')[1].split('.json')[0])
			time = date.getTime()
		} catch (err) {
			return
		}

		if (!date) {
			return
		}

		files.push({
			file,
			date,
			time
		})
	})

	files.sort(compare)

	let combinedData = []

	process.stdout.write('\n')

	files.forEach(day => {
		const filePath = path.join(dataDir, day.file)
		const contents = fs.readFileSync(path.join(__dirname, filePath)).toString()
		const data = JSON.parse(contents)

		
		data.forEach(col => {
			var item = {
				'Time': col[0],
				'Open': col[1],
				'High': col[2],
				'Low': col[3],
				'Close': col[4],
				'Volumn_BTC': col[5],
				'Volumn_Currency': col[6],
				'Weighted_Price': col[7]
			}
			writeDDB(item)
		})

		process.stdout.write('.')
	})

	const outputData = JSON.stringify(combinedData)
})


function writeDDB(item) {
	
	AWS.config.update({
		region: "us-east-2",
		endpoint: "http://localhost:8080"
	  });
	  
	  var docClient = new AWS.DynamoDB.DocumentClient();
	  
	  var table = "BitcoinHistoryData";
	  
	  var params = {
		  TableName:table,
		  Item: item
	  };

	  
	  docClient.put(params, function(err, data) {
		  if (err) {
			  console.error("Unable to add item. Error JSON:", JSON.stringify(err, null, 2));
		  } else {
			  console.log("Added item:", JSON.stringify(data, null, 2));
		  }
	  });
}

File: write_table.py
import boto3
from get_data import get_cols_of_multipe_days
from datetime import date, timedelta, datetime

def write_item_to_ddb(item):
    ddb = boto3.resource('dynamodb')
    table = ddb.Table('BitcoinHistoryData')


    response = table.put_item(
        Item = item
    )


def create_item_for_ddb(col):
    if len(col) < 8:
        print 'Error: unexpected column.'
        return
    item = {}
    item['Time'] = str(col[0])
    item['Open'] = str(col[1])
    item['High'] = str(col[2])
    item['Low'] = str(col[3])
    item['Close'] = str(col[4])
    item['Volumn_BTC'] = str(col[5])
    item['Volumn_Currency'] = str(col[6])
    item['WeightedPrice_USD '] = str(col[7])

    return item

def bulk_write_to_ddb(start_date=date(2010, 1, 1), end_date=date(2050, 12, 31)):
    columns = get_cols_of_multipe_days(start_date, end_date)

    for col in columns:
        try:
            print 'Writing %s' % datetime.fromtimestamp(float(col[0])).isoformat()
            write_item_to_ddb(create_item_for_ddb(col))
        except:
            pass

if __name__ == "__main__":
    bulk_write_to_ddb()